onnxruntime/core/providers/cuda/cu_inc/bitmask.cuh

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "core/providers/cuda/shared_inc/fast_divmod.h"
+
+/**
+ * These functions MUST be called with an unroll factor that evenly divides the number of threads in a warp (32).
```

Comment:
does this always hold for ROCM? https://rocmdocs.amd.com/en/latest/Programming_Guides/HIP-porting-guide.html?highlight=warp#warpsize